first cut of service and app cr's that split broker and application

gtully opened the pull request:
provisioning roles. leaverage restricted mode artemis cr.

brusdev reviewed (commented):
This is a great feature that makes provisioning messaging apps on Kubernetes easy!

gtully committed:
[NO-ISSUE] fix up sample yaml
fix string in config/samples/broker_activemqartemisservice_v1beta1_cr.yaml
fix map notation in config/samples/broker_activemqartemisservice_v1beta1_cr.yaml
fix map notation in config/samples/broker_activemqartemisapp_v1beta1_cr.yaml
Co-authored-by: Bruscino Domenico Francesco <bruscinodf@gmail.com>

gtully committed:
[NO-ISSUE] wrap and report err once on reconcile failure

gtully committed:
[#1186] add validation on required tls secrets for restricted mode

gtully committed:
[NO-ISSUE] add round trip tutorial

Co-authored-by: Thomas Lavocat <tlavocat@redhat.com>

gtully committed:
[#1192] mount an empty broker xml to ensure reload of properties in restricted mode

gtully committed:
[#1180] validate pub sub use case

dependabot[bot] committed:
[NO-ISSUE] Bump actions/checkout from 5 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 5 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

gtully committed:
[NO-ISSUE] set persistent true to use real disk for tutorials to avoid using transient memory

gtully committed:
[#1180] move acceptor from service to app

gtully committed:
[NO-ISSUE] Revert increase logging on markdown-runner to diagnose flaky ci

This reverts commit a33b2ed56faf18fc4c27c9d9d9b0967aacb1cd35.

gtully committed:
[#1180] remove some unnecessary example files, tutorials have it covered

gtully committed:
[#1180] remove matching on auth type, mtls is mandatory

gtully committed:
[#1180] fix typo in file name

gtully committed:
[#1180] qualify kind service with broker, BrokerService

gtully committed:
[#1180] integrate into reconcile loop for update, add unit test

gtully committed:
[#1180] qualify kind app with broker, BrokerApp

gtully committed:
[#1180] remove old crd

gtully committed:
[#1180] remove use of snapshot image now that 2.50.0 released

gtully committed:
[#1180] remove finalizer usage and move app reconcile logic into broker service controller as it owns the app secret that is the source of truth for the service config

gtully committed:
[NO-ISSUE] fix microdnf update issue - nothing provides glibc-common

gtully committed:
[#1180] requeue reconcile when pod status is pending

gtully committed:
[#1180] fix rbac annotations

gtully committed:
[NO-ISSUE] Revert fix microdnf update issue - nothing provides glibc-common

This reverts commit 63ec1b48c30deb83c9bdb2eea0c0d3fdb29cc45d.

gtully committed:
[#1180] MVP of BrokerService and BrokerApp CRs

gtully committed:
[#1180] update test to reflect realm qualifier in jaas config keys

gtully committed:
[NO-ISSUE] correct pod afinity and security context resources

gtully committed:
[#1180] add bindings secret and ref in the status to expose app host and url

[#1180] track cr status as a reconciler instance var

gtully committed:
[#1180] improve status conditions

Assisted-by: Claude,Gemini

gtully committed:
[#1180] implement capacity check on memory limits

Assisted-by: Claude,Gemini

gtully committed:
[#1180] implement cross ns shared address round trip test

Assisted-by: Claude

gtully committed:
[#1180] add broker service controller metric to track provisioned apps

Assisted-by: Claude

lavocatt reviewed (commented):
That's great work!

I've caught a few things that could get improved:

1. Documentation: especially on the business critical sections such as scheduling and user facing bits
2. Code extraction: getting the common broker properties/secrets generation and handling in a common package to reduce the scattering of fmt.println functions accross the codebase
3. Bin-packing: consolidate the scheduling of the apps onto the services

Let me know if you want to discuss these items further.

gtully committed:
[#1180] queue metrics for provisioned apps
